opensrp-opd/src/main/java/org/smartregister/opd/widgets/OpdMultiSelectDrugPicker.java

```diff
@@ -0,0 +1,119 @@
+package org.smartregister.opd.widgets;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.support.annotation.NonNull;
+import android.support.v7.app.AlertDialog;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import com.vijay.jsonwizard.domain.MultiSelectItem;
+import com.vijay.jsonwizard.widgets.MultiSelectListFactory;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.smartregister.opd.R;
+
+import java.util.Arrays;
+
+public class OpdMultiSelectDrugPicker extends MultiSelectListFactory implements TextWatcher {
+    private Context context;
+    private Button opd_btn_save_drug;
+
+    @Override
+    protected void handleClickEventOnListData(@NonNull MultiSelectItem multiSelectItem, @NonNull Context context) {
+        this.context = context;
+        createAdditionalDetailsDialog(multiSelectItem);
+    }
+
+    private void createAdditionalDetailsDialog(@NonNull final MultiSelectItem multiSelectItem) {
+        LayoutInflater inflater = LayoutInflater.from(context);
+        View view = inflater.inflate(R.layout.dosage_and_duration_layout, null);
+        ImageView imgClose = view.findViewById(R.id.multiSelectListCloseDialog);
+        opd_btn_save_drug = view.findViewById(R.id.opd_btn_save_drug);
+
+        TextView txtSelectedItemInMultiSelectList = view.findViewById(R.id.txtSelectedItemInMultiSelectList);
+        txtSelectedItemInMultiSelectList.setText(multiSelectItem.getKey());
+        final EditText edtTreatmentDuration = view.findViewById(R.id.edtTreatmentDosage);
+        final EditText edtTreatmentDosage = view.findViewById(R.id.edtTreatmentDuration);
+        edtTreatmentDosage.addTextChangedListener(this);
+        edtTreatmentDuration.addTextChangedListener(this);
+        AlertDialog.Builder builder = new AlertDialog.Builder(context, R.style.FullScreenDialogStyle);
+        builder.setView(view);
+        builder.setCancelable(true);
+        final AlertDialog alertDialog = builder.create();
+        imgClose.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                alertDialog.dismiss();
+            }
+        });
+        alertDialog.show();
+
+        opd_btn_save_drug.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String duration = edtTreatmentDuration.getText().toString();
+                String dosage = edtTreatmentDosage.getText().toString();
```

Comment:
These two lines will cause a crash because a `null` can be returned by `getText()`. This will happen because both of these fields are not required.

<img width="486" alt="Screen Shot 2019-10-30 at 11 01 22 AM" src="https://user-images.githubusercontent.com/31766075/67839444-bebcdc80-fb04-11e9-9a33-22fc7161ac09.png">
<img width="403" alt="Screen Shot 2019-10-30 at 11 01 32 AM" src="https://user-images.githubusercontent.com/31766075/67839448-c2506380-fb04-11e9-91ab-361c139fd0b4.png">
<img width="485" alt="Screen Shot 2019-10-30 at 11 01 37 AM" src="https://user-images.githubusercontent.com/31766075/67839455-c5e3ea80-fb04-11e9-8d7b-3bd7257c0adb.png">